src/llmcompressor/modifiers/quantization/quantization/base.py

```diff
@@ -213,7 +262,51 @@ def _calibrate_if_possible(self, module: Module):
         elif not self.calibration_dataloader_:
             return
 
+        module.apply(lambda model: initialize_observer(model, base_name="input"))
+        module.apply(lambda model: initialize_observer(model, base_name="output"))
+        module.apply(self.register_calibration_hooks)
         self._calibrate(module)
+        module.apply(set_unset_kv_cache)
+        for h in self.calibration_hooks_:
+            h.remove()
+
+    def register_calibration_hooks(self, module: Module):
+        """
+        Register hooks for input/output activation or kv_cache quantization.
+        """
+        quantization_scheme = getattr(module, "quantization_scheme", None)
+        if not quantization_scheme:
+            return
+
+        is_attention_module_ = is_attention_module(module)
+        input_quant = quantization_scheme.input_activations
+        output_quant = quantization_scheme.output_activations
+
+        calibrate_inputs = (
+            input_quant and not is_attention_module_ and not input_quant.dynamic
+        )
+
+        # Calibrate inputs if an input_quant is provided and not running dynamic quant
+        if calibrate_inputs:
+            pre_hook_handle = module.register_forward_pre_hook(calibrate_input_hook())
+            self.calibration_hooks_.append(pre_hook_handle)
+
+        if output_quant:
+            output_callable_ = None
+            # hooks for attn modules if running kv_cache
+            if is_attention_module_:
+                pre_hook_handle = module.register_forward_pre_hook(
+                    calibrate_kv_cache_input_hook(), with_kwargs=True
+                )
+                self.calibration_hooks_.append(pre_hook_handle)
```

Comment:
i think we should log what gets appended in the hook on debug and when its used on debug